### DISCUSSION TITLE
Checkstyle and Kotlin

### DISCUSSION THREAD

**Discussion by ManosMachairas** [checkstyle/checkstyle#14266] [Q&A] [answered]
Hi! I am currently reading about checkstyle and how to use it and just found this article, which helps you setup Checkstyle in Android projects ([https://medium.com/fantageek/how-to-setup-android-projects-8c306aaeb519]). I know that Checkstyle doesn't support kotlin files, but in this article I bumped into the following line of code "include '**/*.kt'.
![image](https://github.com/checkstyle/checkstyle/assets/46612530/d1141e35-5c01-4ff4-89e5-455fdc2f70f8)

Could someone explain to me what the meaning of this line?

**Comment by nrmancuso** [comment, marked as answer]
Hi @ManosMachairas, please note that we do not maintain the gradle plugin; however, as a gradle user I can tell you that the part of the configuration that you are inquiring about (`include '**/*.kt'`) does indeed explicitly include kotlin files in in the checkstyle task execution. I am not able to read the full article that you've linked in your post, but it is possible to check non-java files with Checkstyle, with the limitation that they are all treated as text files essentially. This means that while you can use checks for regular expressions, line endings, files ending with a newline, etc, you cannot use any java specific checks. If you would like to know more about the checkstyle gradle plugin, you would be better served by inquiring at https://github.com/gradle/gradle.

**Reply by ManosMachairas** [reply]
Thank you for your assistance and fast response!
